src/blockdeque.jl

```diff
@@ -0,0 +1,152 @@
+"""
+    append!(dest::BlockVector, sources...; alias = false)
+
+Append items from `sources` to the last block of `dest`.  If `alias = true`,
+append the blocks in `sources` as-is if they are compatible with the internal
+block array type of `dest`.  Importantly, this means that mutating `sources`
+afterwards alters the items in `dest` and it may even break the invariance
+of `dest` if the length of `sources` are changed.  The elements may be copied
+even if `alias = true` if the corresponding implementation does not exist.
+
+The blocks in `dest` must not alias with `sources` or components of them.
+For example, the result of `append!(x, x)` is undefined.
+"""
+Base.append!(dest::BlockVector, sources...; alias::Bool = false) =
+    foldl((d, s) -> append!(d, s; alias = alias), sources; init = dest)
+
+function Base.append!(dest::BlockVector, src; alias::Bool = false)
+    if _blocktype(dest) === _blocktype(src) && alias
+        return append_nocopy!(dest, src)
+    else
+        return append_copy!(dest, src)
+    end
+end
+
+_blocktype(::Any) = nothing
+_blocktype(::T) where {T<:AbstractArray} = T
+_blocktype(::BlockArray{<:Any,<:Any,<:AbstractArray{T}}) where {T<:AbstractArray} = T
+_blocktype(::PseudoBlockArray{<:Any,<:Any,T}) where {T<:AbstractArray} = T
+
+function append_nocopy!(dest::BlockVector{<:Any,T}, src::BlockVector{<:Any,T}) where {T}
```

Comment:
This adds a new block right? I would call this `blockappend!`, and add a `blockpop!`, etc.

Reply:
Renamed it and added a docstring. Please check if the behavior I described is OK.

It actually does not always add a single block. If `src` is a block vector, it can add multiple blocks.

(Also, I haven't added `blockpop!` etc. yet.)